### DISCUSSION TITLE
Shovel cannot connect on Azure Kubernetes when set up using the messaging topology operator

### DISCUSSION THREAD

**Discussion by paywagh11** [rabbitmq/rabbitmq-server#10223] [Other]
### Describe the bug

I have total 2 rabbitmq (ha)-  3 node rabbitmq cluster running in my Azure kubernetes cluster in 2 different namespaces.  

When I tried to create a shovel through UI I am able to create it and it works as well.

But when I create a yaml file and try to create a shovel, in source rmq it shows that shovel name but not in running state and only in starting state. I used URI the same way I used while creating through UI. Just encoded values of source and destination uri in yaml shovel secret file.
Uri: for source and destination
amqp://username:oassword@hostname/vhost

I refered below git link to create yaml files for shovel
https://github.com/rabbitmq/messaging-topology-operator/blob/main/docs/examples/shovels/README.md

### Reproduction steps

1.create a rmq operator and  3 node rabbitmq cluster in 2 namespaces in Azure kubernetes.
2. Get the username, password and host from rmq secret available in kubernetes cluster for both rmq.
3.create queues, vhosts, shovel secret and shovel using yaml files available at below GitHub link. Update the namespace and rabbitmq cluster reference as per the created ones.
https://github.com/rabbitmq/messaging-topology-operator/blob/main/docs/examples/shovels/README.md
4. Shovel should get created at source rmq but in starting state.
...

### Expected behavior

Shovel should be in running state

**Comment by michaelklishin** [comment]
@paywagh11 we cannot suggest much without seeing any Shovel errors and your actual YAML (without credentials and other sensitive values). [This README](https://github.com/rabbitmq/messaging-topology-operator/blob/main/docs/examples/shovels/README.md) only provides a basic structure.

So start with inspecting server logs. If a Shovel cannot connect, it will log the exception(s) it runs into. In fact, at `debug` level it will log quite a bit of information for any connection, failing or not.

A Shovel in `starting` state tried to connect and failed.

One very common beginner mistake is to try to use the [default user](https://rabbitmq.com/access-control.html#loopback-users) for remote connections. That won't work by design.

**Comment by michaelklishin** [comment]
Another wild guess I have is that the your password contains characters that need escaping in YAML, or possibly on the command line, but do not require extra escaping when used in the management UI form.
